ai_edge_torch/generative/examples/stable_diffusion/pipeline.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2024 The AI Edge Torch Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import os
+from pathlib import Path
+from typing import Dict, Optional
+
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+
+import ai_edge_torch.generative.examples.stable_diffusion.samplers as samplers
+from ai_edge_torch.generative.examples.stable_diffusion.tokenizer import Tokenizer  # NOQA
+import ai_edge_torch.generative.examples.stable_diffusion.util as util
+from ai_edge_torch.model import TfLiteModel
+
+
+class StableDiffusion:
+
+  def __init__(
+      self,
+      *,
+      tokenizer_vocab_dir: str,
+      clip_ckpt: str,
+      encoder_ckpt: Optional[str] = None,
+      diffusion_ckpt: str,
+      decoder_ckpt: str
+  ):
+    self.tokenizer = Tokenizer(tokenizer_vocab_dir)
+    self.clip = TfLiteModel.load(clip_ckpt)
+    self.decoder = TfLiteModel.load(decoder_ckpt)
+    self.diffusion = TfLiteModel.load(diffusion_ckpt)
+    if encoder_ckpt is not None:
+      self.encoder = TfLiteModel.load(encoder_ckpt)
+
+
+def run_tflite_pipeline(
```

Comment:
can this python pipeline easily map to MediaPipe in the future?

Reply:
Should not be a heavy work, given the stable diffusion is implemented with mediapipe before. We need to map the pipeline to the existing mediapipe graph.